client/src/components/UserInfoForm.tsx

```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
 import { shallowEqual, useDispatch, useSelector } from 'react-redux';
```

Comment:
I'm recommending a significant refactoring of this form and its related `userSlice` to improve performance and avoid unnecessary back-and-forth to the Redux store. Since form validation isn't being triggered until the user clicks the 'Next' button, and therefore since instant feedback isn't being provided to the user as they type in the form fields, it is overkill to update immediately to the Redux store anytime the user types in a field, and to even have individual field updater actions, like `setGender`, `setPreference`, etc.

I would instead recommend having one action, perhaps `setUser`, that is only triggered upon click of the 'Next' button after the fields are validated. You could handle pulling uncontrolled values from the form fields using either `useRef()` or React Hook Form. Please let me know if you need clarification or advice as to how to implement it. Or if you want to talk more about the "why" of it.